File: java/com/purescraps/purehtml/transformers/TransformParams.java
package com.purescraps.purehtml.transformers;

import org.jsoup.nodes.Element;

public class TransformParams {

    private Object val;
    private Element element;
    private final String url;

    // Constructor
    public TransformParams(Object val, Element element, String url) {
        this.val = val;
        this.element = element;
        this.url = url;
    }
    // Getters and Setters
    public Object getVal() {
        return val;
    }

    public void setVal(Object val) {
        this.val = val;
    }

    public Element getElement() {
        return element;
    }

    public void setElement(Element $el) {
        this.element = $el;
    }

    public String getUrl() {
        return url;
    }

}

File: python/purehtml/transformers/transform_params.py
from typing import Any

from bs4 import Tag  # Assuming bs4 is used for the Element equivalent in Python

from purehtml.backend.backend import PureHTMLNode


class TransformParams:
    def __init__(self, val: Any, element: PureHTMLNode, url: str):
        """
        Initialize TransformParams with value, element, and URL.
        :param val: The value to be transformed (can be any type).
        :param element: The BeautifulSoup Tag object representing the HTML element.
        :param url: The URL from which the element originated.
        """
        self.val = val
        self.element = element
        self.url = url

    # Getters and Setters
    def get_val(self) -> Any:
        return self.val

    def set_val(self, val: Any) -> None:
        self.val = val

    def get_element(self) -> PureHTMLNode:
        return self.element

    def set_element(self, element: Tag) -> None:
        self.element = element

    def get_url(self) -> str:
        return self.url
